### DISCUSSION TITLE
Directory does not exist

### DISCUSSION THREAD

**Discussion by FernandoHCFaria** [olahallengren/sql-server-maintenance-solution#675] [Questions & Answers]
We bought a new NAS Server (Asustor) and now every time I try to backup SQL I received the error "Directory does not exist" BUT it exist at the NAS Server. 

Date and time: 2022-08-24 11:53:52
Server: SERVIDOR-GP\SQLEXPRESS
Version: 12.0.6169.19
Edition: Developer Edition (64-bit)
Platform: Windows
Procedure: [master].[dbo].[DatabaseBackup]
Parameters: @Databases = 'GP', @Directory = '\\SERVIDOR-NAS\BackupSQL', @BackupType = 'LOG', @Verify = 'Y', @CleanupTime = NULL, @CleanupMode = 'AFTER_BACKUP', @Compress = 'Y', @CopyOnly = 'N', @ChangeBackupType = 'N', @BackupSoftware = NULL, @CheckSum = 'Y', @BlockSize = NULL, @BufferCount = NULL, @MaxTransferSize = NULL, @NumberOfFiles = NULL, @MinBackupSizeForMultipleFiles = NULL, @MaxFileSize = NULL, @CompressionLevel = NULL, @Description = NULL, @Threads = NULL, @Throttle = NULL, @Encrypt = 'N', @EncryptionAlgorithm = NULL, @ServerCertificate = NULL, @ServerAsymmetricKey = NULL, @EncryptionKey = NULL, @ReadWriteFileGroups = 'N', @OverrideBackupPreference = 'N', @NoRecovery = 'N', @URL = NULL, @Credential = NULL, @MirrorDirectory = NULL, @MirrorCleanupTime = NULL, @MirrorCleanupMode = 'AFTER_BACKUP', @MirrorURL = NULL, @AvailabilityGroups = NULL, @Updateability = 'ALL', @AdaptiveCompression = NULL, @ModificationLevel = NULL, @LogSizeSinceLastLogBackup = NULL, @TimeSinceLastLogBackup = NULL, @DataDomainBoostHost = NULL, @DataDomainBoostUser = NULL, @DataDomainBoostDevicePath = NULL, @DataDomainBoostLockboxPath = NULL, @DirectoryStructure = '{ServerName}${InstanceName}{DirectorySeparator}{DatabaseName}{DirectorySeparator}{BackupType}_{Partial}_{CopyOnly}', @AvailabilityGroupDirectoryStructure = '{ClusterName}${AvailabilityGroupName}{DirectorySeparator}{DatabaseName}{DirectorySeparator}{BackupType}_{Partial}_{CopyOnly}', @FileName = '{ServerName}${InstanceName}_{DatabaseName}_{BackupType}_{Partial}_{CopyOnly}_{Year}{Month}{Day}_{Hour}{Minute}{Second}_{FileNumber}.{FileExtension}', @AvailabilityGroupFileName = '{ClusterName}${AvailabilityGroupName}_{DatabaseName}_{BackupType}_{Partial}_{CopyOnly}_{Year}{Month}{Day}_{Hour}{Minute}{Second}_{FileNumber}.{FileExtension}', @FileExtensionFull = NULL, @FileExtensionDiff = NULL, @FileExtensionLog = NULL, @Init = 'N', @Format = 'N', @ObjectLevelRecoveryMap = 'N', @ExcludeLogShippedFromLogBackup = 'Y', @DirectoryCheck = 'Y', @StringDelimiter = ',', @DatabaseOrder = NUL...
Version: 2022-01-02 13:58:13
Source: https://ola.hallengren.com
	
Msg 50000, Nível 16, Estado 1, Procedimento DatabaseBackup, Linha 2276 [Linha de Início do Lote 0]
The directory \\SERVIDOR-NAS\BackupSQL does not exist.
	
Msg 50000, Nível 16, Estado 1, Procedimento DatabaseBackup, Linha 2276 [Linha de Início do Lote 0]
The documentation is available at https://ola.hallengren.com/sql-server-backup.html.
	
Date and time: 2022-08-24 11:53:52
	
![image](https://user-images.githubusercontent.com/106595874/186451109-6c66d252-1460-4afe-9b90-72a2d368b560.png)

**Comment by pbarryuk** [comment]
It looks like there is a problem with your directory parameter - UNC paths usually start with two “\” and yours only has a single “\”.

**Comment by FernandoHCFaria** [comment]
No, im using 2 back slashes. You can see it at parameters line below

EXECUTE dbo.DatabaseBackup
@Databases = 'GP',
@Directory = '\\SERVIDOR-NAS\BackupSQL',
@BackupType = 'LOG',
@Verify = 'Y',
@CleanupTime = NULL,
@CheckSum = 'Y',
@LogToTable = 'Y',
@Compress='Y'

I always used it and always worked. Now, with new nas storage this error
happens

**Comment by DBASam01** [comment]
Just wondering if you ever found a solution for this. I am having a similar issue. The backups just stopped working and the error is that the directory doesn't exist. There was a password change, but the service is still running so I don't see how it could be a bad password.

**Comment by olahallengren** [comment]
@DBASam01, I don't know what has occured in your case, but I have this checklist in the Frequently Asked Questions:

How do I back up to a network share?

- Verify that the SQL Server and the network share are in the same domain.
- Run the SQL Server service under a domain account.
- Run the SQL Server Agent service under a domain account.
- Grant the SQL Server and SQL Server Agent service accounts full control of the directory and the network share.
- If you are running the job under a proxy account, the proxy must be a domain account. Add the account to the sysadmin server role and grant it full control of the directory and the network share.
- Back up to a UNC path.

https://ola.hallengren.com/frequently-asked-questions.html

**Reply by DBASam01** [reply]
@olahallengren The process was working for months and then stopped, so I don't think it has anything to do with the items you mention. The infrastructure team says they think it's something on the SQL end. I don't see what could have changed on the SQL side. The accounts had their password updated, but the services are all running and the server has been rebooted, so I don't think it's a bad password.

**Comment by olahallengren** [comment]
@DBASam01, I can't see that it is an issue in SQL Server.

Please share any findings in the comments in this thread.

**Reply by DBASam01** [reply]
@olahallengren  We think it might be an authentication issue as it works when I change from a domain name to an IP address. It's possible that it's a Kerebos issue. We're still investigating as we consider using the IP address to be a less than optimal solution.

**Reply by DBASam01** [reply]
@olahallengren It appears that the Kerberos service account was locked out. The amazing thing is that this was the only thing to break, frankly.

**Reply by olahallengren** [reply]
> @olahallengren It appears that the Kerberos service account was locked out. The amazing thing is that this was the only thing to break, frankly.

Thank you for reporting back. I'm happy that you found out about the issue.
